OpenSim/Examples/DataTable/example.cpp

```diff
@@ -329,6 +369,145 @@ int main() {
+    tsdt_vec6.addTimestamps({0.0, 0.25, 0.50, 0.75, 1.0});
+
+    // Both a timestamp and a row can be added in one call. Multiple timestamps
+    // and rows can be added in one call as well using addTimestampsAndRows().
+    data_real.clear();
+    data_vec3.clear();
+    data_vec6.clear();
+    data_real.assign(3, 20);
+    data_vec3.assign(3, SimTK::Vec3{40, 50, 60});
+    data_vec6.assign(3, SimTK::Vec6{70, 80, 90, 100, 110, 120});
+    tsdt_real.addTimestampAndRow(1.25, data_real);
+    tsdt_vec3.addTimestampAndRow(1.25, data_vec3);
+    tsdt_vec6.addTimestampAndRow(1.25, data_vec6);
+
+    // Timestamps can be changed. Following changes timestamps of all the rows
+    // at once. It is possible to specify a starting row index to change 
+    // timestamps for only a subset of rows. See documentation for details.
+    tsdt_real.changeTimestamps({0.0, 0.2, 0.4, 0.6, 0.8, 1.0});
+    tsdt_vec3.changeTimestamps({0.0, 0.2, 0.4, 0.6, 0.8, 1.0});
+    tsdt_vec6.changeTimestamps({0.0, 0.2, 0.4, 0.6, 0.8, 1.0});
+
+    // Labeling of columns is exactly same as that for DataTable_.
+    tsdt_real.setColumnLabels({"col-one", "col-two", "col-three"});
+    tsdt_vec3.setColumnLabels({"col-one", "col-two", "col-three"});
+    tsdt_vec6.setColumnLabels({"col-one", "col-two", "col-three"});
+
+    // Rows can be indexed on timestamps.
+    {
+        auto dtrow_real = tsdt_real.getRowOfTimestamp(0.6); ignore(dtrow_real);
+        auto dtrow_vec3 = tsdt_vec3.getRowOfTimestamp(0.6); ignore(dtrow_vec3);
+        auto dtrow_vec6 = tsdt_vec6.getRowOfTimestamp(0.6); ignore(dtrow_vec6);
+
+        tsdt_real.updRowOfTimestamp(0.4) *= 2;
+        tsdt_vec3.updRowOfTimestamp(0.4) *= 2;
+        tsdt_vec6.updRowOfTimestamp(0.4) *= 2;
+    }
+
+    // Timestamps used to index the rows need not exist in the timestamp
+    // column. A row with an approximate match can be obtained as follows.
+    {
+        using OpenSim::NearestDir::LessThanEqual;
+        using OpenSim::NearestDir::GreaterThanEqual;
+        using OpenSim::NearestDir::LessOrGreaterThanEqual;
+
+        // Get a row whose timestamp is less than or equal to given timestamp.
+        // Row 2 (3rd row) is returned for below call.
+        auto dtrow_real = tsdt_real.getRowOfTimestamp(0.5, 
+                                                      LessThanEqual);
+        ignore(dtrow_real);
+        // Get a row whose timestamp is greater than or equal to the given
+        // timestamp. Row 3 (4th row) is returned for below call.
+        auto dtrow_vec3 = tsdt_real.getRowOfTimestamp(0.5,
+                                                      GreaterThanEqual);
+        ignore(dtrow_vec3);
+        // Get a row whose timestamp is closest in either direction to the
+        // given timestamp. Row 3 (4th row) is returned for below call.
+        auto dtrow_vec6 = tsdt_real.getRowOfTimestamp(0.5,
+                                                      LessOrGreaterThanEqual);
```

Comment:
I don't know how amenable these enums are to python scripting. Even if it worked it'd feel unnatural. It seems simpler to have methods like this

```
getRowOfTimestamp
getRowBelowTimestamp
...
```

The enum could be used internally if the implementation of all those methods is very similar.